frontend/javascripts/dashboard/publication_details_view.tsx

```diff
@@ -1,116 +1,72 @@
-import * as React from "react";
+import React, { useState, useEffect } from "react";
 import { Layout, Spin, Tooltip } from "antd";
 import { ArrowLeftOutlined } from "@ant-design/icons";
-import { getDatasets } from "admin/admin_rest_api";
-import type { APIDataset, APIMaybeUnimportedDataset } from "types/api_flow_types";
+import { getPublication } from "admin/admin_rest_api";
+import type { APIPublication } from "types/api_flow_types";
 import PublicationCard from "dashboard/publication_card";
 import { handleGenericError } from "libs/error_handling";
 import { Link } from "react-router-dom";
 const { Content } = Layout;
-type Props = {
-  publicationId: string;
-};
-type State = {
-  datasets: Array<APIMaybeUnimportedDataset>;
-  isLoading: boolean;
-};
-export function SimpleHeader() {
-  return (
-    <div id="oxalis-header">
-      <img
-        src="/assets/images/oxalis.svg"
-        alt="webKnossos Logo"
-        style={{
-          verticalAlign: "middle",
-        }}
-      />
-      webKnossos
-    </div>
-  );
-}
 
-class PublicationDetailView extends React.PureComponent<Props, State> {
-  state: State = {
-    datasets: [],
-    isLoading: true,
-  };
+function PublicationDetailView({ publicationId }: { publicationId: string }) {
+  const [publication, setPublication] = useState<APIPublication | null>(null);
+  const [isLoading, setIsLoading] = useState(false);
 
-  componentDidMount() {
-    this.fetchData();
-  }
+  useEffect(() => {
+    (async () => {
+      try {
+        setIsLoading(true);
+        setPublication(await getPublication(publicationId));
+      } catch (error) {
+        handleGenericError(error as Error);
+      } finally {
+        setIsLoading(false);
+      }
+    })();
+  }, [publicationId]);
 
-  async fetchData(): Promise<void> {
-    try {
-      this.setState({
-        isLoading: true,
-      });
-      const datasets = await getDatasets();
-      this.setState({
-        datasets,
-      });
-    } catch (error) {
-      handleGenericError(error as Error);
-    } finally {
-      this.setState({
-        isLoading: false,
-      });
-    }
-  }
-
-  render() {
-    const { isLoading, datasets } = this.state;
-    const datasetsOfPublication = datasets.filter(
-      (ds) =>
-        ds.isActive && ds.publication != null && ds.publication.id === this.props.publicationId,
```

Comment:
the filtering by `isActive` isn't done anywhere, anymore, right? isn't it still necessary?

Reply:
Good question. @leowe can non-active datasets be part of a publication?

Reply:
The publication simply has a foreign key on the annotations which itself has a foreign key on the datasets. There are no checks in the code on active/non-active datasets except that we don't show deleted datasets.

Reply:
I added an `isActive` filter.